domain/autocert/state/state.go

```diff
@@ -98,13 +99,15 @@ func (st *State) Delete(ctx context.Context, name string) error {
 		return errors.Trace(err)
 	}
 
-	q := `DELETE FROM autocert_cache WHERE name = ?`
+	stmt, err := sqlair.Prepare(`DELETE FROM autocert_cache WHERE name = $dbAutocert.name`, dbAutocert{})
+	if err != nil {
+		return errors.Annotatef(err, "preparing autocert cache delete statement")
+	}
 
-	err = db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
-		if _, err := tx.ExecContext(ctx, q, name); err != nil {
-			return errors.Trace(err)
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		if err := tx.Query(ctx, stmt, dbAutocert{Name: name}).Run(); err != nil {
```

Comment:
You might as well create this earlier, and provide it to `Prepare` instead of instantiating a new empty one there.

Reply:
I've changed it to this, though I'm not a huge fan of passing variables to Prepare rather than just new instantiated structs. I think it can make it look like there is a reason the variable has been instantiated the way it has when really we only care about the type.